app/controllers/api/v3/sync_controller.rb

```diff
@@ -31,6 +30,20 @@ def __sync_to_user__(response_key)
 
   private
 
+  def merge_records(params)
+    params.flat_map { |single_entity_params|
+      result = merge_if_valid(single_entity_params)
+      AuditLog.merge_log(current_user, result[:record]) if result[:record].present?
+      result
+    }
+  end
+
+  # This is a hook method for subclasses that need to operate on the results after the merge but before the response
+  # is sent back to the client.
+  # results will be an array of Hashes, with the key values of either { :record => [Model] } or { :errors_hash => Hash }
+  def after_merge_completed(results)
+  end
+
```

Comment:
We can remove all the changes in this controller and the `after_merge_completed` template method as well, right? That was only needed for rollups?